opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -363,6 +363,49 @@ private ExprValue parseArray(
     return new ExprCollectionValue(result);
   }
 
+  /**
+   * Parse geo point content.
+   *
+   * @param content Content to parse.
+   * @param supportArrays Parsing the whole array or not
+   * @return Geo point value parsed from content.
+   */
+  private ExprValue parseGeoPoint(Content content, boolean supportArrays) {
+    // there is only one point in doc.
+    if (content.isArray() == false) {
+      final var pair = content.geoValue();
+      return new OpenSearchExprGeoPointValue(pair.getLeft(), pair.getRight());
+    }
+
+    var elements = content.array();
+    var first = elements.next();
+    // an array in the [longitude, latitude] format.
```

Comment:
@dai-chen Actually, I have commented on one format. The other formats are handled by `GeoUtils#parseGeoPoint`.